frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx

```diff
@@ -143,14 +186,32 @@ const createCsiVaultResources = (
     VAULT_CLIENT_CERT: kms.clientCert?.metadata.name,
     VAULT_CLIENT_KEY: kms.clientKey?.metadata.name,
     VAULT_NAMESPACE: kms.providerNamespace,
-    VAULT_TOKEN_NAME: kms.token.value ? KMSSecretName : '',
     VAULT_CACERT_FILE: kms.caCertFile,
     VAULT_CLIENT_CERT_FILE: kms.clientCertFile,
     VAULT_CLIENT_KEY_FILE: kms.clientKeyFile,
+    VAULT_AUTH_METHOD: kms.authMethod,
   };
+
+  switch (kms.authMethod) {
+    case VaultAuthMethods.TOKEN:
+      // encryption using vault token
+      csiConfigData.KMS_PROVIDER = KmsImplementations.VAULT_TOKENS;
+      break;
+    case VaultAuthMethods.KUBERNETES:
+      // encryption using tenant serviceAccount
+      csiConfigData.KMS_PROVIDER = KmsImplementations.VAULT_TENANT_SA;
+      csiConfigData = {
+        ...csiConfigData,
+        VAULT_AUTH_PATH: kms.providerAuthPath,
+        VAULT_AUTH_NAMESPACE: kms.providerAuthNamespace,
+      };
+      break;
+    default:
+  }
+
```

Comment:
IIRC, when going thr wizard flow:
if cluster-wide is checked + Vault token --> we create both configMaps (ocs-kms and csi-kms) and secret for Rook
if storage-class is checked + Vault token --> we create both configMaps and secret for Ceph-CSI 
@GowthamShanmugam please let me know if I am correct ??

Also, can u plz point me where we are craeting secret for ceph-csi in `openshift-storage` NS

Reply:
or along with both configMaps we are creating both secrets as well ?

Reply:
yes we always create both configMap, and secret creation for  Ceph-CSI  on openshift-storage NS is not this PR, you can find that logic in: https://github.com/openshift/console/pull/10563